### PR TITLE
BlendMode diagrams

### PR DESCRIPTION
**Pull request by Hixie**
cc @gspencergoog 

this depends on https://github.com/flutter/assets-for-api-docs/pull/16 and https://github.com/flutter/flutter/pull/13987.

These are intended to be used by new docs for the BlendMode enum, not yet written.

### CODE REVIEW

File: src/dart-ui/blend_mode.dart [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      new RRect.fromRectXY(rect, rect.shortestSide / 3.0, rect.shortestSide / 3.0),
+      paint,
+    );
+  }
+
+  @override
+  bool shouldRepaint(Diagram oldDelegate) {
+    return mode != oldDelegate.mode;
+  }
+}
```
Comment (gspencergoog) [inline src/dart-ui/blend_mode.dart]: Add newline.

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by Hixie** [comment]
Can you find a cute vertical-aspect-ratio partially-transparent CC0 image I can use instead? :-)

**Comment by Hixie** [comment]
found one. gonna improve these further and resubmit.

**Comment by Hixie** [comment]
I uploaded the changes as a second commit if you want to review them.

I'll wait to land thus until after the cleanup one is reviewed and landed, since it depends on it.

**Comment by Hixie** [comment]
Also depends on https://github.com/flutter/flutter/pull/14009.

**Pull request merged**